src/cloudai/schema/test_template/nemo_launcher/slurm_install_strategy.py

```diff
@@ -293,3 +294,22 @@ def _clone_repository(self, subdir_path: str) -> None:
         result = subprocess.run(checkout_cmd, cwd=repo_path, capture_output=True, text=True)
         if result.returncode != 0:
             raise RuntimeError(f"Failed to checkout commit: {result.stderr}")
+
+    def _install_requirements(self, subdir_path: str) -> None:
+        """
+        Installs the required Python packages from the requirements.txt file in the cloned repository.
+
+        Args:
+            subdir_path (str): Subdirectory path for installation.
+        """
+        repo_path = os.path.join(subdir_path, self.REPOSITORY_NAME)
+        requirements_file = os.path.join(repo_path, "requirements.txt")
+
+        if os.path.isfile(requirements_file):
+            logging.debug("Installing requirements from %s", requirements_file)
+            install_cmd = ["pip", "install", "-r", requirements_file]
```

Comment:
Isn't it risky to call `pip` without knowing if virtual env is created? It might pollute main python env or fail if we don't have permission to change it.

Reply:
Yes, it makes sense. To update the behavior, we need to update how NeMo is installed. Currently, it is reusing the venv created for CloudAI. About 4-5 months ago, I tried to create a separate venv for NeMo and use it, but this approach did not work well, so I reverted to the current method. Let's stick with this approach for now to fix the bug, and then move on to a better solution.